serverside-javascript-best-practices.md

```diff
@@ -320,6 +350,54 @@ createBooking.call(self, product, params, 'upgrades', 'supplier', 'payment', fun
 
 
 
+## Handling heavily sequential async code with promises
+
+* !! Using promises for heavily sequential async code has the downside of hiding the data being passed between functions. We lose the ability to wrap each layer in a closure to selectively pass the correct data between the correct entites. Everything must flow through the middle to get to the end. This breaks our rules on single-responsibility, but allows for slightly cleaner orchestration:
+* !! Sinon promises do not support sandboxing yet, so you'll have to manually restore everything by hand
+* !! It's impossible to prove that the data flow between promises is correct, only that data moves between them.
```

Comment:
I think some code demonstrating these issues would be useful. Particularly, where you've said `"it's impossible"`, `"we lose the ability"` and `"everything must"` - I think it's important developers understand. 

For example I'm not sure why it's _"impossible to prove the flow of data between promises is correct"_. I'm wondering why we can't we just spy on them and see what they're called with and what they resolve or don't resolve with?